_scripts/git-precommit-hook.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# 1. Get the list of all modified and dependent packages.
+# Cleans up yarn's output by removing everything before the package name, the first and the last line.
+PACKAGES_MODIFIED=`yarn workspaces list --since=HEAD^ -R | cut -d/ -f2 | sed '1d' | sed '$d'`
```

Comment:
## Test cases I tried

### Passed
* Only fxa root files modified
  * No building done
* Fxa root files and one package modified
  * Builds the modified package and any packages that depend on it successfully.
* One package modified
  * Builds the modified package and any packages that depend on it successfully.
* Multiple packages modified with overlapping packages that depend on them.
  * Builds the modified package and any packages that depend on it once each successfully.
* No files modified (e.g. `git commit --amend` to reword a commit)
  * No building done
* Multiple packages modified where one has a build error.
  * Error message logged to console and build logs accessible via `./artifacts/build-modified-and-dependent.log`.

### Failed
* Modified package with build error, but it is not staged.
  * No building done.
  * AFAICT [`yarn workspaces list`](https://yarnpkg.com/cli/workspaces/list) doesn't have a `--staged` option. I'm passing `HEAD^` as the git ref in the `--since` option, but this will pick up anything that has changed.
  * This is a less common scenario in my experience, but I'd like to get that working too if possible -- any ideas?